### PR TITLE
Feat: add query stac for automate cron workflow

### PR DESCRIPTION
**Pull request by lhoupert**
Main Changes:

- Adds a new Python script scripts/query_stac.py that queries a STAC API to identify new items in a source collection within a configurable time window
- The script checks if items have already been processed in a target collection and outputs a JSON list of items that need processing
- Includes comprehensive unit tests (tests/unit/test_query_stac.py) with mock STAC clients covering various scenarios including error handling and edge cases
- Adds minimal JSON fixtures for reproducible testing of source and target collections

Contributes to implementation of  first phase _1. Recent Data Processing (Hourly)_  of https://github.com/EOPF-Explorer/coordination/issues/181,  as discussed in ADR https://github.com/EOPF-Explorer/coordination/pull/179

### CODE REVIEW

File: scripts/query_stac.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    sys.stderr.write("\n📊 Summary:\n")
+    sys.stderr.write(f"  Checked: {checked_count} items\n")
+    sys.stderr.write(f"  To process: {len(items_to_process)} items\n\n")
+    sys.stderr.flush()
+
+    # Output ONLY JSON to stdout (for Argo withParam)
+    sys.stdout.write(json.dumps(items_to_process))
+    sys.stdout.flush()
+
```
Comment (emmanuelmathot) [inline scripts/query_stac.py]: Use logger instead

**Review by emmanuelmathot — APPROVED** [review]
